### ISSUE TITLE
Problems with call_mods

### ISSUE THREAD

**Issue by lucio-cyber** [bioinfomaticsCSU/deepsignal#57]
Hi Peng,
I ran the following command: "deepsignal call_mods --input_path fast5s.al.CpG.signal_features.17bases.rawsignals_360.tsv --model_path model.CpG.R9.4_1D.human_hx1.bn17.sn360.v0.1.7 + / bn_17.sn_360 .epoch_9.ckpt --result_file fast5s.al.CpG.call_mods.tsv --nproc 20 --is_gpu no "and deepsignal tells me:
"## parameters:
input_path:
fast5s.al.CpG.signal_features.17bases.rawsignals_360.tsv
model_path:
model.CpG.R9.4_1D.human_hx1.bn17.sn360.v0.1.7 + / bn_17.sn_360.epoch_9.ckpt
kmer_len:
17
cent_signals_len:
360
batch_size:
512
learning_rate:
0.001
class_num:
2
result_file:
fast5s.al.CpG.call_mods.tsv
recursively:
yes
corrected_group:
RawGenomeCorrected_000
basecall_subgroup:
BaseCalled_template
reference_path:
It is not
is_dna:
yes
normalize_method:
mad
methy_label:
1
motifs:
CG
mod_loc:
0
f5_batch_num:
100
nproc:
20
is_gpu:
no
# ==============================================
write_process started ..
2020-09-13 19: 30: 03.756002: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 04.019482: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 04.191046: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 04.342153: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 04.887242: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 05.001080: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 05.661715: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 05.842513: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 06.000198: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 06.043159: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 06.129178: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 06.411155: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 07.016522: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 07.065846: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 07.616329: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-13 19: 30: 08.002258: W tensorflow / core / framework / op_kernel.cc: 1318] OP_REQUIRES failed at save_restore_v2_ops.cc:184: Not found: Key BDGRU_rnn / bw / multi_rnn_cell / cell_0 / lstm_cell / bias not found in checkpoint
2020-09-13 19: 30: 08.088356: W tensorflow / core / framework / op_kernel.cc: 1318] OP_REQUIRES failed at save_restore_v2_ops.cc:184: Not found: Key BDGRU_rnn / bw / multi_rnn_cell / cell_0 / lstm_cell / bias not found in checkpoint
2020-09-13 19: 30: 08.214039: W tensorflow / core / framework / op_kernel.cc: 1318] OP_REQUIRES failed at save_restore_v2_ops.cc:184: Not found: Key BDGRU_rnn / bw / multi_rnn_cell / cell_0 / lstm_cell / bias not found in checkpoint
2020-09-13 19: 30: 08.220377: W tensorflow / core / framework / op_kernel.cc: 1318] OP_REQUIRES failed at save_restore_v2_ops.cc:184: Not found: Key BDGRU_rnn / bw / multi_rnn_cell / cell_0 / lstm_cell / bias not found in checkpoint
2020-09-13 19: 30: 08.413802: I tensorflow / core / platform / cpu_feature_guard.cc: 140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA

He gave me the file.tsv but I'm not sure he executed the command correctly because it made me the errors mentioned above in a very short time.

Other exception like : " # pylint: disable=protected-access"

What can it mean?

I installed tensorflow as indicated in the instructions in a virtual environment. The current version of tensorflow is 1.8.0.
I have not installed the one for the GPU because this is not that powerful and then in the "call_mods" command I set "--is_gpu no".
What can it be due to then?
What can I do?

**Comment by PengNi** [comment]
Can you check the version of deepsignal by using `pip list`? It seems that the version of deepsignal is lower than 0.1.7.

**Comment by lucio-cyber** [comment]
Yes indeed it is, it tells me that the version of deepsignal is 0.1.4

**Comment by PengNi** [comment]
You can re-install or update deepsignal to 0.1.7:
```shell
pip install deepsignal==0.1.7 
```

Best,
Peng

**Comment by lucio-cyber** [comment]
Hi Peng,
I upgraded Deepsignal but it always gives me the same thing.
I now have the following versions:
Tensorflow == 1.8.0
Tensorflow-estimator == 2.3.0
Deepsignal == 0.1.7

**Comment by PengNi** [comment]
Can you use `which deepsignal` to see where the bin file `deepsignal` is? May be the bin file is still the 1.4.0 one.

I suggest that you uninstall deepsignal and then re-install it.

Best,
Peng

**Comment by lucio-cyber** [comment]
I did not understand your question. So should I just uninstall deepsignal or also tensorflow and tombo and then reinstall them all?

**Comment by PengNi** [comment]
Just deepsignal.

**Comment by lucio-cyber** [comment]
Always the version you suggested before, deepsignal 0.1.7 with the command “pip install deepsignal == 0.1.7” or just “pip install deepsignal”?

**Comment by PengNi** [comment]
Using
```
pip uninstall deepsignal
# then
pip install deepsignal==0.1.7
```
Also there is no space in the model path:
```
model.CpG.R9.4_1D.human_hx1.bn17.sn360.v0.1.7+/bn_17.sn_360.epoch_9.ckpt
```

**Comment by lucio-cyber** [comment]
What do you mean by "There is also no space in the model path"

**Comment by PengNi** [comment]
What I mean are that make sure: 1. reinstall deepsignal; 2. the command for calling methylation is right.

Best,
Peng

**Comment by lucio-cyber** [comment]
Ok! Thank you so much Peng for helping me, I will try that and I will let you know. Thanks again.
Best Regards,
Lucio

**Comment by PengNi** [comment]
1. `--prob_cf` is for ignoring ambiguous calls. If in a call `abs(prob0 - prob1)>prob_cf`, we abandon the call. However, `--prob_cf` is experimental args, we didn't test the best value.

2. Either `call_modification_frequency.py` or `call_modification_frequency2.py` is ok to use. `call_modification_frequency2.py` allows muti call_mods files as input in `append` mode.

Best,
Peng
